File: Python/3. 스택/10828.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/10828
'''

import sys
input = sys.stdin.readline # 입력 속도를 빠르게 만들기 위해 꼭 쓴다.

n = int(input())
stack = []
for i in range(n) :
    a = input().split()
    if a[0] == 'push' : 
        stack.append(a[1])
    elif a[0] == 'pop' : 
        if len(stack) == 0 :
            print(-1)
        else :
            print(stack[-1])
            stack.pop()
    elif a[0] == 'size' : 
        print(len(stack))
    elif a[0] == 'empty' :
        if len(stack) == 0 :
            print(1)
        else :
            print(0)
    else :
        if len(stack) == 0 :
            print(-1) 
        else :
            print(stack[-1])

File: C++/7. 스택 - 초급/10828.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/10828
*/

#include <iostream>
#include <stack>

using namespace std;

int n, t;
stack<int> s;

int main(){

    cin >> n;
    
    for(int i=0; i<n; i++){
        string str;
        cin >> str;
        if(str == "push"){
            cin >> t;
            s.push(t);
        }else if(str == "pop"){
            if(!s.empty()){
                cout << s.top() << "\n";
                s.pop();
            }else{
                cout << "-1\n";
            }
        }else if(str == "size"){
            cout << s.size() << "\n";
        }else if(str == "empty"){
            cout << s.empty() << "\n";
        }else{
            if(!s.empty()){
                cout << s.top() << "\n";
            }else{
                cout << "-1\n";
            }
        }
    }
    return 0;
}